### PR TITLE
Btm 227 index transaction data

### PR DESCRIPTION
**Pull request by MelAtGDS**

### CODE REVIEW

**Comment by MelAtGDS** [comment]
I can't approve as I opened the PR but I've read through it. Definitely don't understand all of the cloudformation stuff but nothing jumped out at me.

File: cloudformation/athena.yaml [hunk 1]
```diff
@@ -102,3 +102,58 @@ Resources:
...
+        CloudWatchEncryption:
+          CloudWatchEncryptionMode: SSE-KMS
+          KmsKeyArn: !GetAtt KmsKey.Arn
+        JobBookmarksEncryption:
+          JobBookmarksEncryptionMode: CSE-KMS
+          KmsKeyArn: !GetAtt KmsKey.Arn
+        S3Encryptions:
+          - S3EncryptionMode: SSE-KMS
+            KmsKeyArn: !GetAtt KmsKey.Arn
+      Name: transactions-crawler-security-config
```
Comment (phinterseer) [inline cloudformation/athena.yaml:144]: Is this name going to conflict with other stacks? If in doubt, always use the prefix for names.

File: cloudformation/athena.yaml [hunk 2]
```diff
@@ -102,3 +102,58 @@ Resources:
...
+
+  TransactionsCrawler:
+    Type: AWS::Glue::Crawler
+    Properties:
+      CrawlerSecurityConfiguration: !Ref TransactionsCrawlerSecurityConfiguration
+      DatabaseName: !Ref TransactionDB
+      Description: Crawls the daily buckets of transaction data
+      Name: btm-transactions-crawler
```
Comment (phinterseer) [inline cloudformation/athena.yaml:152]: Is this name going to conflict with other stacks? If in doubt, always use the prefix for names.

File: cloudformation/global.yaml [hunk 3]
```diff
@@ -27,6 +27,7 @@ Resources:
                   - s3.amazonaws.com
                   - sns.amazonaws.com
                   - sqs.amazonaws.com
+                  - logs.eu-west-2.amazonaws.com
```
Comment (phinterseer) [inline cloudformation/global.yaml:30]: Why does this entry have a region and the others don't?

**Review by phinterseer — DISMISSED** [review]
See comments

**Review by phinterseer — APPROVED** [review]

File: integration_tests/helpers/s3Helper.ts [hunk 4]
```diff
@@ -7,9 +7,10 @@ import {
 } from "@aws-sdk/client-s3";
 import { ReadStream } from "fs";
 
-async function getS3ItemsList(bucketName: string) {
+async function getS3ItemsList(bucketName: string, bucketPrefix: string) {
```
[the remark this answers is not in the document]
Comment (NKannan14) [reply integration_tests/helpers/s3Helper.ts:10]: @markschnitziusgovuk  already reviewed the auto tests changes but no issues you can review as well  :)

File: cloudformation/calculation.yaml [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+      PointInTimeRecoverySpecification:
+        PointInTimeRecoveryEnabled: true
+      ProvisionedThroughput:
+        ReadCapacityUnits: 5
+        WriteCapacityUnits: 5
+      SSESpecification:
+        KMSMasterKeyId: !GetAtt KmsKey.Arn
+        SSEEnabled: true
+        SSEType: KMS
+
```
[the remark this answers is not in the document]
Comment (markschnitziusgovuk) [reply cloudformation/calculation.yaml:45]: Removed

File: integration_tests/helpers/s3Helper.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (markschnitziusgovuk) [reply integration_tests/helpers/s3Helper.ts:10]: Going to resolve this just to get it merged...  Nithya is okay with doing suggested changes later.

**Pull request merged**